authz/rbac_translator.go

```diff
@@ -94,7 +94,8 @@ func getStringMatcher(value string) *v3matcherpb.StringMatcher {
 	switch {
 	case value == "*":
 		return &v3matcherpb.StringMatcher{
-			MatchPattern: &v3matcherpb.StringMatcher_SafeRegex{},
+			MatchPattern: &v3matcherpb.StringMatcher_SafeRegex{
+				SafeRegex: &v3matcherpb.RegexMatcher{Regex: ".*"}},
```

Comment:
In SDK authorization policy, `*` value represents presence match i.e. it will match when the value is not empty.

Here `.*` regex would match anything, including empty strings.
I will update this to regex for matching non empty strings (`^\\S+$`)

But I think there is an issue with underlying code.
https://github.com/grpc/grpc-go/blob/master/internal/grpcutil/regex.go#L26
https://pkg.go.dev/regexp#Regexp.FindString

FindString returns empty string if there is no match.
So when the value to match is an empty string, and we use the right regex and get failed match.
Then https://github.com/grpc/grpc-go/blob/master/internal/grpcutil/regex.go#L27 it returns true (instead of the expected failed match)

I think we should use the alternatives mentioned in the link (like FindStringIndex or FindStringSubmatch).

@dfawley Let me know your thoughts. Thank you!

Reply:
Seems like the `FullMatchWithRegex` function there may need to call `re.Match`?  E.g. `return len(rem) == len(string) && re.Match(string)`.  Or optimize and only call `Match` if `len(string) == 0`.  @zasweq @menghanl 

Also, can we _not_ use `string` as a variable name, please?

Reply:
@dfawley @zasweq @menghanl Made the updates to `FullMatchWithRegex` in this PR. PTAL